wiki/Gameplay/Daily_challenge/en.md

```diff
@@ -0,0 +1,61 @@
+# Daily Challenge
+
+The **daily challenge** is a multiplayer mode in osu!(lazer) where players are able to build up a daily streak by passing beatmaps consecutively with increasing difficulty with the difficulty being reset every 7 days.
+
+Every beatmap is handpicked from the [Featured Artist](/wiki/People/Featured_Artists) listing, and sometimes the maps come with a mod forced onto users, which requires them to pass the map with the selected mod.
+
+## Game menu
+
+From the main menu, the daily challenge menu can be accessed with the following steps:
+
+1. Click the `Play` button or press `P`.
+2. Click the `Daily challenge` button or press `D`.
+
+![](img/daily-challenge-menu.png "Screenshot of the daily challenge menu during November 7, 2024.") 
+
+On entering, the user is presented with an intro showing the beatmap the user has to pass, and the mods they have to pass it with.
+
+After the intro ends, the user can view the data for today's challenge. The left side displays various score-related information like the total pass count and the cumulative total score. In the middle, a leaderboard shows the top scores achieved by players. Users are able to discuss the daily challenge in the chat on the right.
+
+## Beatmap difficulty range
+
+![](img/Beatmap-selection-criteria.png "Specific star ratings beatmaps have to follow for the entire week")
+
+## Streak milestones
+
+Play streaks are displayed on the user profile with different colours depending on certain streak milestones.
+
+### Rank breakdown 
```

Comment:
would be superb if you showcase the colors in the table.

what I'm thinking is having a screenshot of the daily challenge streak text in each row with its respective color.

should be pretty straightforward to make with some HTML editing. let me know if you need help with this otherwise.

Reply:
> should be pretty straightforward to make with some HTML editing.

I'm too bad with JS and using commands in the console overall to do this, so I would appreciate some help 😅 (I'm new to even HTML, so don't judge me *too* much)

Tho in what way are you planning to show the colours? Are you going to input the day/week requirement in the console and later replace the requirement text with the screenshot, or do you want to do this for only a specific type of streak, or an even more different way?
I feel like replacing the text with the screenshots completely would feel tacky, but we can try that anyway I suppose.

Reply:
made this, can send over my changes if you want.

![image](https://github.com/user-attachments/assets/9edbffcb-08c7-4d56-9f7b-23f54291ba04)

Reply:
It's weird they are all offset by one day. That wasn't the intention. If this is an error with implementation we should fix that first?

Reply:
it's just this line: https://github.com/ppy/osu-web/blob/b24f634812725d35bf720f344f58536990360022/resources/js/profile-page/daily-challenge.tsx#L25

Reply:
well we either change that or change this page to read `>1080` etc. it looks so weird atm

Reply:
I've looked into this on lazer/web, and adjusting the checks in https://github.com/ppy/osu/blob/master/osu.Game/Overlays/Profile/Header/Components/DailyChallengeStatsTooltip.cs#L148 to use `>=` for the checks would result in even weirder numbers for total participation and weekly streak because of *rounding*, such as 1079d counting as Lustrous in total participation.

Would be better to just use `>1080` here for now until we do more consideration for the current implementation.

Reply:
I don't see why rounding should come into the equation. That sounds like a bug regardless of the off-by-one consideration?

Reply:
I've opened PRs to attempt having expected and consistent results across web and client for this, while explaining the issue there:

https://github.com/ppy/osu-web/pull/11698
https://github.com/ppy/osu/pull/30906

Reply:
since one of the pr's was already merged, I'm assuming we just change the requirements in the table to match the new calculation.
Tho, do you still want me to add a `>` to the numbers? I can't see a reason to do that, but id rather ask than to push changes already.

Reply:
will apply with the new batch of changes, closing.